src/tbp/monty/frameworks/models/evidence_matching/learning_module.py

```diff
@@ -718,6 +723,7 @@ def _update_possible_matches(self, query):
         # NOTE: would not need to do this if we are still voting
         # Call this update in the step method?
         self.possible_matches = self._threshold_possible_matches()
+        self.previous_mlh = self.current_mlh
         self.current_mlh = self._calculate_most_likely_hypothesis()
```

Comment:
This is more of an FYI, since I don't think it makes sense to do it here with such long variable and function names, but another way to write this that you'll see often in Python code is to do both assignments at the same time.
```suggestion
        self.previous_mlh, self.current_mlh = self.current_mlh, self._calculate_most_likely_hypothesis()
```